Source/Dafny/Plugins/Configuration.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Dafny.Plugins;
+
+/// <summary>
+/// A class that plugins should extend, in order to receive plugin-specific command-line arguments
+///
+/// If no class extending Microsoft.Dafny.Plugins.Configuration is defined in the assembly plugin,
+/// Dafny will implicitly load all classes extending Rewriter
+/// providing them with an ErrorReporter as the first and only argument
+/// </summary>
+public abstract class Configuration {
+  /// <summary>
+  /// A Microsoft.Dafny.Plugins.Configuration will be automatically instantiated an arguments
+  /// will be provided to the plugin by the method `ParseArguments``;
```

Comment:
This is a bit under-specified: is a plugin intended to document which additional arguments it handles?

Would it make more sense to include arguments in the same way that `/verificationLogger:trx;LogFileName=...` works? That way you don't have to worry about conflicts across plugins. You could share the parsing logic here: https://github.com/dafny-lang/dafny/blob/master/Source/DafnyDriver/BoogieXmlConvertor.cs#L35-L46

Reply:
Now I realize that you ARE accepting lists of plugin-scoped arguments instead already. :) But my format suggestion still stands for more consistency, and being able to pass "key=value" arguments is probably smoother.

Reply:
Okay, fair point that it's more flexible to let each plugin decide how to parse arguments.